lib/icingadb/redisconnection.cpp

```diff
@@ -527,3 +574,15 @@ void RedisConnection::WriteOne(RedisConnection::Query& query, asio::yield_contex
 void RedisConnection::SetConnectedCallback(std::function<void(asio::yield_context& yc)> callback) {
 	m_ConnectedCallback = std::move(callback);
 }
+
+void RedisConnection::IncreasePendingQueries(int count)
+{
+	m_PendingQueries += decltype(m_PendingQueries)(count);
+	m_InputQueries.InsertValue(Utility::GetTime(), count);
+}
+
+void RedisConnection::DecreasePendingQueries(int count)
+{
+	m_PendingQueries -= decltype(m_PendingQueries)(count);
+	m_OutputQueries.InsertValue(Utility::GetTime(), count);
+}
```

Comment:
Why these casts instead of just also making `count` an `uint(_fast)64_t`? Also avoids `size_t` -> `int` -> `uint_fast64_t` conversion. Or just use `size_t` all the way. The queue can't grow bigger than that and I better hope that your CPU is fast at handling this type.